src/pymatgen/io/vasp/outputs.py

```diff
@@ -345,7 +345,7 @@ def __init__(
                 self.update_potcar_spec(parse_potcar_file)
                 self.update_charge_from_potcar(parse_potcar_file)
 
-        if self.incar.get("ALGO") not in {"CHI", "BSE"} and not self.converged and self.parameters.get("IBRION") != 0:
+        if self.incar.get("ALGO") not in {"Chi", "Bse"} and not self.converged and self.parameters.get("IBRION") != 0:
```

Comment:
Looks like before migrating to `UserDict`, the `proc_val` is not triggered through `update` method when init from a dict (instead of str), therefore these values remain as it when it should be capitalized.